src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -403,11 +397,27 @@ class BaseCouplingScheme : public CouplingScheme {
    */
   bool accelerate();
 
+  /**
+   * @brief stores current data in buffer of Waveform
+   */
+  void storeWindowData();
+
   /**
    * @brief Extrapolate coupling data from values of previous time windows
-   * @param data Data fields to extrapolate
    */
-  void extrapolateData(DataMap &data);
+  void extrapolateData();
+
+  /**
+   * @brief used for storing send/receive data at end of acceleration, if not converged.
+   */
+  void storeLastIteration() // @todo: Try to make this private. This method is currently, only used for sake of testing.
+  {
+    PRECICE_ASSERT(isImplicitCouplingScheme());
+    for (DataMap::value_type &pair : getAccelerationData()) {
+      // PRECICE_ASSERT(pair.second->lastIteration.size() > 0);  // @todo: Why is this assertion failing?
```

Comment:
TODO

Reply:
This assertion does not make sense anymore. `CouplingData::_lastIteration` (renamed to `_previousIteration`) is uninitialized at the beginning and only gets initialized with `CouplingData::values()` when calling `CouplingData::storeIteration()` the first time. Removed it in dd75e064837846d3da99deed1f8ed12ea6e841db.